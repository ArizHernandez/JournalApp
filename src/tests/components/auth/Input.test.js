const { shallow } = require("enzyme")
const { Input } = require("../../../components/auth/Input")

describe('Pruebas en el <Input />', () => {
  const handleInputChange = jest.fn();
  const wrapper = shallow(
    <Input 
      required={true} 
      autoComplete="false"
      labelText="Nombre:"
      placeholder="Name"
      name="name"
      type="text" 
      value=""
      handleInputChange={handleInputChange}
      expresionRegular={/^.{5,40}$/}
      errorMessage="Nombre no valido"  
    />
  )

  test('Debe de cargar correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('Debe de llamar la funciÃ³n handleInputChange', () => {
    wrapper.find('input').simulate('change');
    
    expect(handleInputChange).toHaveBeenCalled();
  })
  
  test('No debe de mostrar el mensaje de error', () => {
    expect(wrapper.find('.auth__error-message').exists()).toBe(false);
  })
})