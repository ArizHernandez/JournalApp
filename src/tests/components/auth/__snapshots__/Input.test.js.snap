// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <Input /> Debe de cargar correctamente 1`] = `
<div
  className="mb-2"
>
  <label
    className="mb-1"
  >
    <small
      className="
          false
          
        "
    >
      Nombre:
    </small>
  </label>
  <div
    className="auth__input-container"
  >
    <input
      autoComplete="false"
      className="
            false 
            
            auth__input
          "
      name="name"
      onChange={[MockFunction]}
      onKeyUp={[Function]}
      placeholder="Name"
      required={true}
      type="text"
      value=""
    />
  </div>
</div>
`;
